lib/charms/kafka/v0/kafka_snap.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""## Overview
+
+`kafka_snap.py` provides a collection of common functions for managing snap installation and
+config parsing common to both the Kafka and ZooKeeper charms
+
+"""
+
+import logging
+from typing import Dict, List
+
+from ops.model import StatusBase
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import snap
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
+
+logger = logging.getLogger(__name__)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "73d0f23286dd469596d358905406dcab"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 4
+
+
+class KafkaSnap:
+    def __init__(self) -> None:
+        self.default_config_path = "/snap/kafka/current/opt/kafka/config/"
+        self.snap_config_path = "/var/snap/kafka/common/"
+        self.kafka = snap.SnapCache()["kafka"] 
+
+    def install_kafka_snap(self) -> StatusBase:
+        """Loads the Kafka snap from LP, returning a StatusBase for the Charm to set."""
```

Comment:
missing `Returns` in the doc string
also, shouldn't the return type be `ActiveStatus`? (`MaintenanceStatus` indicates a long running job in the charm, but the long running snap installation is already complete)

Reply:
After the completion of `install` there are still stages to complete before `ActiveStatus`, (e.g starting the service, validating config, loading config etc) so I'm just passing back Maintenance with a more informative message for a user (i.e `XYZ failed/succeeded`)